src/org/thoughtcrime/securesms/mms/AttachmentManager.java

```diff
@@ -352,6 +353,10 @@ public void onReceivedDuration(int millis) {
           height = dimens.second;
         }
 
+        if (fileName == null) {
+          fileName = new File(uri.getPath()).getName();
```

Comment:
`uri.getPath()` may return null, in this case, `new File()` will throw an error.

i suggest to add a try/catch block around this statement.

Reply:
Ough, you're right. I'm not used to looking for this kind of problem anymore after programming Rust and Kotlin a lot :wink:

Does a try/catch block have any advantage over just checking for null?

Reply:
in this case, checking for `null` would probably do the job - esp. as we just checked the possible return values.

but one is not always that accurate, called function may be adapted, implementations can change - and `getPath()` and `getName()` might also throw an error in the future. what, if getPath() returns a path `new File()` cannot handle?

so, assuming, that there are many things that can go wrong in uri to file conversion, i would go for a try/catch block for exactly this added line (although i am not the biggest fan of try/catch :) 

also, try/catch on that line makes sure we do not get a regression by this fix.

Reply:
Ok.